tests/src/JIT/Performance/CodeQuality/Span/Indexer.cs

```diff
@@ -0,0 +1,1021 @@
+                }
+
+                Console.WriteLine("Running as command line perf test: {0} iterations",
+                    CommandLineInnerIterationCount);
+                Console.WriteLine();
+            }
+            else
+            {
+                Console.WriteLine("Running as correctness test: {0} iterations",
+                    CommandLineInnerIterationCount);
+                Usage();
+            }
+
+            // When we call into Invoke, it'll need to know this isn't xunit-perf running
+            IsXunitInvocation = false;
+
+            // Discover what tests to run via reflection
+            TypeInfo t = typeof(IndexerBench).GetTypeInfo();
+
+            var testsByCategory = new Dictionary<string, List<MethodInfo>>();
+
+            // Do a first pass to find out what categories of benchmarks we have.
+            foreach(MethodInfo m in t.DeclaredMethods)
+            {
+                BenchmarkAttribute benchAttr = m.GetCustomAttribute<BenchmarkAttribute>();
+                if (benchAttr != null)
+                {
+                    string category = "none";
+                    CategoryAttribute categoryAttr = m.GetCustomAttribute<CategoryAttribute>();
+                    if (categoryAttr != null)
+                    {
+                        category = categoryAttr.Name;
+                    }
+
+                    List<MethodInfo> tests = null;
+
+                    if (!testsByCategory.ContainsKey(category))
+                    {
+                        tests = new List<MethodInfo>();
+                        testsByCategory.Add(category, tests);
+                    }
+                    else
+                    {
+                        tests = testsByCategory[category];
+                    }
+
+                    tests.Add(m);
+                }
+            }
+
+            foreach(string categoryName in testsByCategory.Keys)
+            {
+                Console.WriteLine("**** {0} ****", categoryName);
+
+                foreach(MethodInfo m in testsByCategory[categoryName])
+                {
+                    // Request a warm-up iteration before measuring this benchmark method.
+                    DoWarmUp = true;
+
+                    // Get the benchmark to measure as a delegate taking the number of inner-loop iterations to run
+                    var invokeMethod = m.CreateDelegate(typeof(Action<int>)) as Action<int>;
+
+                    // All the benchmarks methods in this test use [InlineData] to specify how many times and with
+                    // what arguments they should be run.
+                    foreach (InlineDataAttribute dataAttr in m.GetCustomAttributes<InlineDataAttribute>())
+                    {
+                        foreach (object[] data in dataAttr.GetData(m))
+                        {
+                            // All the benchmark methods in this test take a single int parameter
+                            invokeMethod((int)data[0]);
```

Comment:
This calls a [Benchmark] method which in turn calls Invoke, and one of the arguments to Invoke is a string with the method name and argument. We have the MethodInfo and argument handy here, so we could compute the string here and communicate it to Invoke via a static field, to avoid the need to have that boilerplate string creation in each [Benchmark] method. I wouldn't stop to do that in this change, just wanted to mention the idea so that if we repeat this pattern again we could consider doing it then.